packages/amplication-server/src/core/module/module.service.ts

```diff
@@ -223,6 +228,109 @@ export class ModuleService extends BlockTypeService<
       entityId
     );
 
+    await this.validateDtoReferencesBeforeDeleteModule(moduleId, resourceId);
+    await this.validateActionsDtosReferencesBeforeDeleteModule(
+      moduleId,
+      resourceId
+    );
+
     return super.delete({ where: { id: moduleId } }, user, true); //delete the module and all its children (actions/type...)
   }
+
+  private async validateDtoReferencesBeforeDeleteModule(
+    moduleId: string,
+    resourceId: string
+  ) {
+    const allResourceModuleDtos = await this.moduleDtoService.findMany({
+      where: {
+        resource: {
+          id: resourceId,
+        },
+      },
+    });
+
+    const moduleModuleDtos = allResourceModuleDtos.filter(
+      (moduleDto) => moduleDto.parentBlockId === moduleId
+    );
+
+    const allOtherResourceCustomModuleDtos = allResourceModuleDtos.filter(
+      (moduleDto) =>
+        moduleDto.parentBlockId !== moduleId &&
+        (moduleDto.dtoType === EnumModuleDtoType.Custom ||
+          moduleDto.dtoType === EnumModuleDtoType.CustomEnum)
+    );
+
+    allOtherResourceCustomModuleDtos.forEach((moduleDto) => {
+      moduleDto.properties.forEach((prop) => {
+        if (prop) {
+          prop.propertyTypes.forEach((propType) => {
+            const currentDto = moduleModuleDtos.find(
+              (x) => x.id === propType.dtoId
```

Comment:
what if this property is a primitive and not a DTO? you would not have propType.dtoId (maybe that's okay in this case actually, but we need to make sure)

Reply:
@mulygottlieb, In this case, it will not have the dtoId field

Reply:
but don't you need at least `propType?.dtoId` or will it just be an existing field with `undefined` value?

Reply:
An existing field with an undefined value